packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/TypeFactory.cs

```diff
@@ -12,32 +12,102 @@
 
 namespace Microsoft.Generator.CSharp
 {
-    public abstract class TypeFactory
+    public class TypeFactory
     {
         private ChangeTrackingListDefinition? _changeTrackingListProvider;
         private ChangeTrackingListDefinition ChangeTrackingListProvider => _changeTrackingListProvider ??= new();
 
         private ChangeTrackingDictionaryDefinition? _changeTrackingDictionaryProvider;
         private ChangeTrackingDictionaryDefinition ChangeTrackingDictionaryProvider => _changeTrackingDictionaryProvider ??= new();
 
-        private Dictionary<InputType, TypeProvider>? _csharpToTypeProvider;
-        private Dictionary<InputType, TypeProvider> CSharpToTypeProvider => _csharpToTypeProvider ??= [];
+        private Dictionary<InputType, TypeProvider?>? _csharpToTypeProvider;
+        private Dictionary<InputType, TypeProvider?> CSharpToTypeProvider => _csharpToTypeProvider ??= [];
 
         private Dictionary<InputType, CSharpType>? _typeCache;
         private Dictionary<InputType, CSharpType> TypeCache => _typeCache ??= [];
 
-        private Dictionary<InputModelProperty, PropertyProvider>? _propertyCache;
-        private Dictionary<InputModelProperty, PropertyProvider> PropertyCache => _propertyCache ??= [];
+        private HashSet<InputType>? _nullTypes;
+        private HashSet<InputType> NullTypes => _nullTypes ??= [];
+
+        private Dictionary<InputModelProperty, PropertyProvider?>? _propertyCache;
+        private Dictionary<InputModelProperty, PropertyProvider?> PropertyCache => _propertyCache ??= [];
+
+        private Dictionary<InputParameter, ParameterProvider?>? _parameterCache;
+        private Dictionary<InputParameter, ParameterProvider?> ParameterCache => _parameterCache ??= [];
 
         private Dictionary<InputType, IReadOnlyList<TypeProvider>>? _serializationsCache;
+        private IList<LibraryVisitor> Visitors => (IList<LibraryVisitor>)CodeModelPlugin.Instance.GetLibraryVisitors();
         private Dictionary<InputType, IReadOnlyList<TypeProvider>> SerializationsCache => _serializationsCache ??= [];
 
-        public CSharpType CreateCSharpType(InputType inputType)
+        public CSharpType? CreateCSharpType(InputType inputType)
+        {
+            if (NullTypes.Contains(inputType))
+            {
+                return null;
+            }
+
+            CSharpType? type = CreateCSharpTypeCore(inputType);
+
+            if (type == null)
+            {
+                NullTypes.Add(inputType);
+            }
+
+            return type;
+        }
+
+        private protected virtual CSharpType? CreateCSharpTypeCore(InputType inputType)
+        {
+            CSharpType? type;
+            switch (inputType)
+            {
+                case InputLiteralType literalType:
```

Comment:
I think literal can stay in primitive, can't have a literal of a model.

Reply:
It can be a literal of an enum